test/testlib/minikube_utilities.go

```diff
@@ -315,6 +316,32 @@ func AwaitDatabaseUp(t *testing.T, namespace string, podName string, databaseNam
 		"--timeout", "300")
 }
 
+func GetDiagnoseOnTestFailure(t *testing.T, namespace string, podName string) {
+	if t.Failed() && shouldGetDiagnose() {
+		options := k8s.NewKubectlOptions("", "")
+		options.Namespace = namespace
+
+		targetDirPath := filepath.Join(RESULT_DIR, namespace, "diagnose")
+		_ = os.MkdirAll(targetDirPath, 0700)
+
+		// Get cores
+		// Once DB-29847 is implemented, we can set a --timeout or --wait-forever flags
+		// So that core dump streams doesn't timeout in minikube environments
+		t.Log(t, "Generating diagnose archive...")
+		k8s.RunKubectl(t, options, "exec", podName, "--", "nuocmd", "get", "diagnose-info",
+			"--include-cores", "--output-dir", "/tmp")
+		diagnoseFile, err := k8s.RunKubectlAndGetOutputE(t, options, "exec", podName, "--", "bash", "-c", "ls -1 /tmp | grep diagnose-")
+		assert.NilError(t, err, "Can not find diagnose archive")
+
+		k8s.RunKubectl(t, options, "cp", podName+":/tmp/"+diagnoseFile, filepath.Join(targetDirPath, diagnoseFile))
+		if shouldPrintToStdout() {
+			// The file can be recovered via xxd -r -p a.hex a.bin
+			output, _ := exec.Command("hexdump", "-ve", `16/1 "%02x " "\n"`, filepath.Join(targetDirPath, diagnoseFile)).CombinedOutput()
+			t.Logf("%s:\n%s", diagnoseFile, string(output))
```

Comment:
I am a bit worried that this dump will be bigger than the allowed Travis log size, but we can deal with that if it happens.

Reply:
Yes ... I hope that we won't need cores all the time. The good thing is that everything goes to the `results` directory at least for local test execution. At some point, there was a talk about S3 storage so maybe we can do it in future if needed.